unit-tests/sw-dev/test-metadata.py

```diff
@@ -0,0 +1,190 @@
+
+        test.check_false( f1.supports_frame_metadata( rs.frame_metadata_value.actual_fps ))
+except:
+    test.unexpected_exception()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Multiple streams per sensor should share metadata" )
+try:
+    with sw.sensor( "Stereo Module" ) as sensor:
+        depth = sensor.video_stream( "Depth", rs.stream.depth, rs.format.z16 )
+        ir = sensor.video_stream( "Infrared", rs.stream.infrared, rs.format.y8 )
+        sensor.start( depth, ir )
+
+        sensor.set( rs.frame_metadata_value.white_balance, 0xbaad )
+        d1 = sensor.publish( depth.frame() )
+
+        sensor.set( rs.frame_metadata_value.actual_fps, 0xf00d )
+        ir1 = sensor.publish( ir.frame() )
+
+        sensor.set( rs.frame_metadata_value.saturation, 0x1eaf )
+        d2 = sensor.publish( depth.frame() )
+
+        sensor.set( rs.frame_metadata_value.contrast, 0xfee1 )
+        ir2 = sensor.publish( ir.frame() )
+
+        sensor.set( rs.frame_metadata_value.contrast, 0x600d )
+        d3 = sensor.publish( depth.frame() )
+        ir3 = sensor.publish( ir.frame() )
+
+        test.check_equal( d1.get_frame_metadata( rs.frame_metadata_value.white_balance ), 0xbaad )
+        test.check( ir1.supports_frame_metadata( rs.frame_metadata_value.white_balance ))
+        test.check_equal( d2.get_frame_metadata( rs.frame_metadata_value.actual_fps ), 0xf00d )
+        test.check_equal( ir2.get_frame_metadata( rs.frame_metadata_value.saturation ), 0x1eaf )
+        test.check_equal( ir2.get_frame_metadata( rs.frame_metadata_value.contrast ), 0xfee1 )
+        test.check_equal( d3.get_frame_metadata( rs.frame_metadata_value.contrast ), 0x600d )
+        test.check_equal( ir3.get_frame_metadata( rs.frame_metadata_value.contrast ), 0x600d )
+except:
+    test.unexpected_exception()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Two sensors, intermixed frames" )
+try:
+    with sw.sensor( "Stereo Module" ) as stereo:
+        depth = stereo.video_stream( "Depth", rs.stream.depth, rs.format.z16 )
+        stereo.start( depth )
+        with sw.sensor( "RGB Camera" ) as rgb:
+            color = sensor.video_stream( "Color", rs.stream.color, rs.format.yuyv )
+            rgb.start( color )
+
+            stereo.set( rs.frame_metadata_value.white_balance, 0xbaad )
+            d1 = stereo.publish( depth.frame() )
+            test.check_equal( d1.get_frame_metadata( rs.frame_metadata_value.white_balance ), 0xbaad )
+
+            rgb.set( rs.frame_metadata_value.actual_fps, 0xf00d )
+            stereo.set( rs.frame_metadata_value.actual_fps, 0xbaad )
+            c1 = rgb.publish( color.frame() )
+            test.check_false( f1.supports_frame_metadata( rs.frame_metadata_value.actual_fps ))
```

Comment:
Did you mean `d1`?

Reply:
I think so...`f1` is from one of the previous tests... :(